src/Runner/FileCachingLintingIterator.php

```diff
@@ -24,13 +24,13 @@
  */
 final class FileCachingLintingIterator extends \CachingIterator
 {
+    private LinterInterface $linter;
+
     /**
      * @var LintingResultInterface
      */
     private $currentResult;
```

Comment:
```suggestion
    private ?LintingResultInterface $currentResult;
```

Reply:
similar for `nextResult`?

Reply:
I'm not 100% convinced these PHPDoc are always correct, so I skip it for now as I don't want to propose changes I don't 100% understand